lib/bootloader_setup.pm

```diff
@@ -882,50 +882,56 @@ sub grep_grub_cmdline_settings {
 
 =head2 change_grub_config
 
-    change_grub_config($old, $new [, $search ] [, $modifiers ]);
+    change_grub_config($old, $new [, $search ] [, $modifiers ], [, $update_grub ]);
 
-Replace $old with $new in /etc/default/grub, using sed.
+Replace C<$old> with C<$new> in /etc/default/grub, using sed.
 C<$search> meant to be for changing only particular line for sed,
 C<$modifiers> for sed replacement, e.g. "g".
+C<$update_grub> if set, update grub config and upload configuration.
 =cut
 sub change_grub_config {
-    die((caller(0))[3] . ' expects from 2 to 4 arguments') unless (@_ >= 2 && @_ <= 4);
-    my ($old, $new, $search, $modifiers) = @_;
+    die((caller(0))[3] . ' expects from 3 to 5 arguments') unless (@_ >= 3 && @_ <= 5);
+    my ($old, $new, $search, $modifiers, $update_grub) = @_;
```

Comment:
I got confused with `update_grub` variable name in function `change_grub_config` . Got faulty impression that if I would call it with `$update_grub = 0` function will not change anything. What about `$grub_mkconfig` name ? 

Also we need to give ability to pass this option also in case when you don't want to bother about `$modifier` value. Right way to do it would be put both variables into hash

Reply:
> I got confused with `update_grub` variable name in function `change_grub_config` . Got faulty impression that if I would call it with `$update_grub = 0` function will not change anything. 

That was the intention, isn't it working? BTW: I added missing `$update_grub //= 0;`

> What about `$grub_mkconfig` name ?

`C<$update_grub> if set, update grub config and upload configuration.`
It's a "flag", that's why I thought $update_grub` is more descriptive.

> Also we need to give ability to pass this option also in case when you don't want to bother about `$modifier` value. Right way to do it would be put both variables into hash

Possible with `change_grub_config($old, $new [, $search ] "", 1);`
I was thinking about hash, but while `$old` and `$new` are mandatory, for hash would be only the 3rd and 4th parameter (`$modifiers` and `$update_grub`).

Reply:
ok